### PR TITLE
Update UserDefaults.swift

### PR DESCRIPTION
**Pull request by severeduck**

### CODE REVIEW

File: Sources/UserDefaults/UserDefaults.swift [hunk 1]
```diff
@@ -26,7 +26,7 @@ import Foundation
 public struct UserDefault<Value: Codable> {
   private let key: String
   private let defaultValue: Value
-  private var container: UserDefaults = .standard
+  private let container: UserDefaults = .standard
```
Comment (EngOmarElsayed) [inline Sources/UserDefaults/UserDefaults.swift:29]: If you made it let, this means that will not be able to edit the container if needed. The container is not constant but rather  a value that can be changed, it is include in the init but as an optional. So needs to be var

**Review by EngOmarElsayed — CHANGES_REQUESTED** [review]
And even the build failed because of this change

File: Sources/UserDefaults/UserDefaults.swift (hunk 1 above)
Comment (severeduck) [reply Sources/UserDefaults/UserDefaults.swift:29]: right, that was my point. i just wanted to ask if it's possible to have an immutable container property inside your structure.

Comment (EngOmarElsayed) [reply Sources/UserDefaults/UserDefaults.swift:29]: Actully, you can if you wrote it like this `private let container: UserDefaults` without giving it a value. And initialize the value from the `init`. I reponed the PR for you edit it if you want @severeduck

**Comment by severeduck** [comment]
> And even the build failed because of this change

I never tried to build it, man. It's just that the `var` felt unnecessary. Best of luck in your development ☭

File: Sources/UserDefaults/UserDefaults.swift (hunk 1 above)
Comment (severeduck) [reply Sources/UserDefaults/UserDefaults.swift:29]: no, YOU edit it if you want :)

Comment (severeduck) [reply Sources/UserDefaults/UserDefaults.swift:29]: because i dont.

Comment (EngOmarElsayed) [reply Sources/UserDefaults/UserDefaults.swift:29]: I thought you want to contribute

Comment (severeduck) [reply Sources/UserDefaults/UserDefaults.swift:29]: >And initialize the value from the `init`

wouldn't that be great? you then pass a closure to the `init`. For your unit tests - you'll only need to pass another (mock/dummy) closure to see how the system reacts.

Comment (severeduck) [reply Sources/UserDefaults/UserDefaults.swift:29]: i did contribute by giving some ideas. I'm not gonna write the code for you.

**Pull request closed without merging**